proto/substrait/algebra.proto

```diff
@@ -444,6 +444,7 @@ message Rel {
     HashJoinRel hash_join = 13;
     MergeJoinRel merge_join = 14;
     NestedLoopJoinRel nested_loop_join = 18;
+    SingleJoinRel single_join = 22;
```

Comment:
Note that this was previously attempted as a separate join type: https://github.com/substrait-io/substrait/blob/7b794379c51436620515b5ebad7720c502aab991/proto/substrait/algebra.proto#L175

We should either deprecate that join type or collapse this PR into better documenting that join type.  I'm uncertain which approach I'd prefer and think it hinges on the "at most one" or "exactly one" question